pkg/loader/compose/compose.go

```diff
@@ -307,72 +307,67 @@ func (c *Compose) LoadFile(files []string) kobject.KomposeObject {
 		logrus.Fatalf("Failed to load compose file: %v", err)
 	}
 
-	// transform composeObject into komposeObject
-	composeServiceNames := composeObject.ServiceConfigs.Keys()
-
 	noSupKeys := checkUnsupportedKey(composeObject)
 	for _, keyName := range noSupKeys {
 		logrus.Warningf("Unsupported %s key - ignoring", keyName)
 	}
 
-	for _, name := range composeServiceNames {
-		if composeServiceConfig, ok := composeObject.ServiceConfigs.Get(name); ok {
-			serviceConfig := kobject.ServiceConfig{}
-			serviceConfig.Image = composeServiceConfig.Image
-			serviceConfig.Build = composeServiceConfig.Build.Context
-			serviceConfig.ContainerName = composeServiceConfig.ContainerName
-			serviceConfig.Command = composeServiceConfig.Entrypoint
-			serviceConfig.Args = composeServiceConfig.Command
-			serviceConfig.Build = composeServiceConfig.Build.Context
-
-			envs := loadEnvVars(composeServiceConfig.Environment)
-			serviceConfig.Environment = envs
-
-			// load ports
-			ports, err := loadPorts(composeServiceConfig.Ports)
-			if err != nil {
-				logrus.Fatalf("%q failed to load ports from compose file: %v", name, err)
-			}
-			serviceConfig.Port = ports
+	for name, composeServiceConfig := range composeObject.ServiceConfigs.All() {
```

Comment:
Below this line code is as it is, only for loop has changed and the unncecessary things above are removed.